extractor/src/main/java/org/schabi/newpipe/extractor/utils/Utils.java

```diff
@@ -44,22 +68,84 @@ public static String removeNonDigitCharacters(String toRemove) {
     public static long mixedNumberWordToLong(String numberWord) throws NumberFormatException, ParsingException {
         String multiplier = "";
         try {
-            multiplier = Parser.matchGroup("[\\d]+([\\.,][\\d]+)?([KMBkmb])+", numberWord, 2);
-        } catch(ParsingException ignored) {}
+            multiplier = Parser.matchGroup("[\\d]+([\\.,][\\d]+)?([KMBkmb万লক億])+", numberWord, 2);
+        } catch (ParsingException ignored) {
+        }
         double count = Double.parseDouble(Parser.matchGroup1("([\\d]+([\\.,][\\d]+)?)", numberWord)
                 .replace(",", "."));
         switch (multiplier.toUpperCase()) {
             case "K":
                 return (long) (count * 1e3);
```

Comment:
`1e3` is more readable than `1000`, but it could cause issues since it is a floating point number. Floats are not perfectly precise, so we could get that `(long) (x * 1e3) != x * 1000`, because `x * 1e3` could be a little bit less (i.e. 0.0000001 less) than `x * 1000` so when converting it to long, that just floors the value, we would get `x * 1000 - 1`. So I think it would be better here and in the following lines to use `1000`, `10000`, `100000`, etc.

Reply:
Ah. I got once while testing a value using  
```java
case "ল": //100K, used by indo-arabic languages
                return (long) (count * 1e5);
```
or 10M, I don't know, with only 1 difference between the English count and the language count (it was like a number, let's say 16M, with English I got 16M, with these languages I got 16M - 1).

I think the best for readability is to use underscores:
https://docs.oracle.com/javase/7/docs/technotes/guides/language/underscores-literals.html